app/helpers/elasticsearch_helper.rb

```diff
@@ -14,13 +14,14 @@ def prefix_match(model, field, prefix, condition)
     results
   end
 
-  def taxon_search(prefix)
+  def taxon_search(prefix, tax_levels = %w[species genus], label_levels = false)
     return {} if Rails.env == "test"
     matching_taxa = {}
-    %w[species genus].each do |level|
+    tax_levels.each do |level|
       search_params = { query: { query_string: { query: "#{prefix}*", fields: ["#{level}_name"] } } }
       TaxonLineage.__elasticsearch__.search(search_params).records.each do |record|
         name = record["#{level}_name"]
+        name += " (#{level})" if label_levels
```

Comment:
Can we return more fields?
And return the level separately, i.e. `level=`? 
The client should handle any kind of concatenation themselves.

Reply:
Thanks, I'll make that change